fuel-p2p/src/service.rs

```diff
@@ -551,6 +550,128 @@ mod tests {
         }
     }
 
+    // Simulate 2 Sets of Sentry nodes.
+    // In both Sets, a single Guarded Node should only be connected to their sentry nodes.
+    // While other nodes can and should connect to nodes outside of the Sentry Set.
+    #[tokio::test]
+    #[instrument]
+    async fn sentry_nodes_working() {
+        let reserved_nodes_size = 4;
+        let double_reserved_nodes_size = reserved_nodes_size * 2;
+
+        let mut p2p_config = P2PConfig::default_with_network("sentry_nodes_working");
+        // enable mdns for faster discovery of nodes
+        p2p_config.enable_mdns = true;
+
+        // let's increase the ideal mesh number
+        // this still should not affect the nodes that are enabling `reserved_nodes_only_mode`
+        let gossipsub_builder = default_gossipsub_builder()
+            .mesh_n(double_reserved_nodes_size)
+            .mesh_n_low(double_reserved_nodes_size - 2)
+            .mesh_n_high(double_reserved_nodes_size + 4)
```

Comment:
We should have a comment explaining why we need the mesh size to be larger than the reserved node size, so that we don't have people making up their own random reasons about the significance of these magic numbers when refactoring or updating these tests in the future.

Reply:
I wrote a more detailed comment, plus I made mesh lows and highs same as the ideal mesh size to avoid any confusion.